enhancements/two-node-fencing/tnf.md

```diff
@@ -242,6 +252,12 @@ The plan is for this to be changed by one of the nodes during pacemaker initiali
 While set to `External`, CEO will still need to render the configuration for the etcd container in a place where it can be consumed by pacemaker. This ensures that the etcd instance managed by pacemaker can be updated
 accordingly in the case of a upgrade event or whenever certificates are rotated.
 
+In case CEO will be used as the in-cluster operator responsible for setting up Pacemaker fencing it'll require root permissions which are currently mandatory to run the required pcs commands.
```

Comment:
Maybe we want to mention that the current goal of this is to have these commands that require root running as a separate operand under CEO. That operand pod will be privileged, but its scope should be limited to just running these pacemaker-related commands.

Reply:
Hmm I'm not sure it would matter as IIUC even if we'll have separate operand running under CEO (which requires root permissions) that would mean that CEO also needs root permissions.

Reply:
> that would mean that CEO also needs root permissions

That's my assumption at least as well. Everything else (a pod without root can create a pod with root) would be a security hole IMHO. I will find out...

Reply:
My thinking was that cri-o creates all of the pods. CEO just creates and applies the manifest for its operands. So I guess the question is if there is an permission set associated with the CEO operator that would restrict what level of privilege CRI-O is allowed to set. We will await your guidance, @slintes :)

Reply:
it's a bit more complicated ;)

- something during OCP installation creates the CEO deployment resource (IIRC it's CVO, cluster version operator?)
- the API server's controller manager creates a replica set, the next controller creates the pod, the scheduler assigns a node to it
- `kubelet` on the assigned node sees the new pod, and uses cri-o for creating the container(s) of the pod
- the code running in the container is creating the operand's deployment: the API server will check though if this is allowed, using the pod's ServiceAccount, RBAC rules and probably some more (SCC?)
- if the API server denies creation: no operand deployment resource, kubelet and cri-o not involved
- if API server allows: goto step 2 :)

Reply:
Based on my own research, permissions for a pod in a namespace depends on the definition of the service account defined in that namespace or explicitly overrideen the the related pod. So the service account used by CEO (or whatever container we would use to create the operand) is key thing we need to look at.
We should probably start by seeing if CEO uses the default service account for its namespace - which only has API discovery permissions. (https://kubernetes.io/docs/concepts/security/service-accounts/#default-service-accounts)\

Reply:
CEO has its own service account [1], and (as I noticed just now) has cluster admin RBAC rules assigned to it [2]
The interesting part now is if cluster admin RBAC rules alone allow creating privileged deployments and mounting host directories, while the CEO deployment itself isn't privileged...
[1] https://github.com/openshift/cluster-etcd-operator/blob/master/manifests/0000_20_etcd-operator_05_serviceaccount.yaml
[2] https://github.com/openshift/cluster-etcd-operator/blob/master/manifests/0000_20_etcd-operator_04_clusterrolebinding.yaml

Reply:
there still is SCC [1] and PSA [2] involved...
[1] https://docs.openshift.com/container-platform/4.17/authentication/managing-security-context-constraints.html
[2] https://kubernetes.io/docs/concepts/security/pod-security-admission/